app/assets/javascripts/components/students/shared/StudentList/StudentDrawer/TrainingStatus/ExerciseRows.jsx

```diff
@@ -16,22 +16,46 @@ const ExerciseStatusCell = ({ status, sandboxUrl }) => {
 // Helper Functions
 const orderByDueDate = (a, b) => (moment(a.due_date).isBefore(b.due_date) ? -1 : 1);
 
-const generateRow = status => (exercise) => {
+// This function returns the status of exercise
+// returns status as unread/complete/incomplete depending on the deadline status and flags if any
+const getExerciseStatus = (exercise) => {
+  const isComplete = exercise.deadline_status === 'complete';
+  const flags = exercise.flags || {};
+  let status = '';
+  if (isComplete && flags.marked_complete) {
+    status = 'complete';
+  } else if (isComplete) {
+    status = 'incomplete';
+  } else {
+    status = 'unread';
+  }
+  return status;
+};
+
+// This function compares exercise's due date with current date
+// returns true if the current date has not passed the training's due date
+const isExerciseDue = (date) => {
```

Comment:
Looks like we do this same calculation in both components. It should be extracted to avoid duplicating the code.

Reply:
Yes, even the sorting function was same for both the files, so I created a new file [trainingHelperFunctions.js](https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/4374/commits/554733e1ea526ca3b94a3786f60cfa9ef16c6d35#diff-10a081ff2907f54569ade38a05ffede3c406925a6af2be5c3e881606ab050ab2) which contains all the helper functions used for rendering trainings and exercises.